drivers/adc/adc_smartbond_sdadc.c

```diff
@@ -119,34 +115,24 @@ static inline void sdadc_smartbond_pm_policy_state_lock_get(const struct device
 					      struct sdadc_smartbond_data *data)
 {
 #if defined(CONFIG_PM_DEVICE)
-#if defined(CONFIG_PM_DEVICE_RUNTIME)
 	pm_device_runtime_get(dev);
-#endif
-
-	if (!atomic_test_and_set_bit(data->pm_policy_state_flag, 0)) {
-		/*
-		 * Prevent the SoC from entering the normal sleep state.
-		 */
-		pm_policy_state_lock_get(PM_STATE_STANDBY, PM_ALL_SUBSTATES);
-	}
+	/*
+	 * Prevent the SoC from entering the normal sleep state.
+	 */
+	pm_policy_state_lock_get(PM_STATE_STANDBY, PM_ALL_SUBSTATES);
 #endif
 }
 
 static inline void sdadc_smartbond_pm_policy_state_lock_put(const struct device *dev,
 					      struct sdadc_smartbond_data *data)
 {
 #if defined(CONFIG_PM_DEVICE)
-#if defined(CONFIG_PM_DEVICE_RUNTIME)
+	/*
+	 * Allow the SoC to enter the normal sleep state once sdadc is done.
+	 */
+	pm_policy_state_lock_put(PM_STATE_STANDBY, PM_ALL_SUBSTATES);
```

Comment:
are you sure that you want this call under this ifdef ? I thought even if you are building without device runtime pm you still want to prevent the SoC goes standby when the device is in use.

Reply:
Sleep should be constrained when PM device is used, either the runtime or system-managed mode. `CONFIG_PM_DEVICE` should reflect both PM device modes, isn't it? If PM is not used, the SoC should not enter any sleep state. Let me know if this is what you meant.

Reply:
Nope, you can have only system power management without device power management. That means that the susbystem will still gets idle and sleeps but devices are never suspended or resumed.